sailor/_base/fetch.py

```diff
@@ -1,15 +1,15 @@
 from itertools import product
 import operator
 from typing import List
-import warnings
 import re
 import logging
 
-from ..utils.oauth_wrapper import get_oauth_client
-from ..utils.utils import DataNotFoundWarning, _is_non_string_iterable
+from sailor.utils.oauth_wrapper import get_oauth_client
+from sailor.utils.utils import DataNotFoundWarning, _is_non_string_iterable, WarningAdapter
 
 LOG = logging.getLogger(__name__)
 LOG.addHandler(logging.NullHandler())
+log_adapter = WarningAdapter(LOG)
```

Comment:
Since the adapter supports all logging functions.. just replace the normal LOG with an instance of the adapter. Should be fine. Saves a line and a variable.

```
LOG = WarningAdapter(logging.getLogger(__name__))
LOG.addHandler(logging.NullHandler())
```
You can do this in all the files where you made changes.

Reply:
The logger adapter delegates most function calls to the logger, but the `addHandler` function is an exception here. Thus it cannot be called with an instance of the adapter. As a result I cannot replace the code as proposed. 
I can think of two options how to handle it: Either keep the code as it is (maybe additionally renaming `log_adapter` to `LOG`) or adding a function `addHandler` to our `WarningAdapter`, which delegates the call to the underlying logger.

Reply:
Alright. Since this is only true if we manage to adapt the stacklevel for all calls in the `loggerAdapter` properly (eg. `adapter.info` and `adapter.error`) I will make this change once the stacklevel issue is solved.

Reply:
done